### DISCUSSION TITLE
How to compile against the SDL backend

### DISCUSSION THREAD

**Discussion by Sqvid** [raysan5/raylib#3764] [Q&A] [answered]
I am currently facing Issue #3763 which is possibly due to the glfw backend. This made me wonder how I could go about linking against SDL instead. Though this was announced in the changelog, there don't seem to be any instructions on how to do this. I would greatly appreciate any help. Thank you.

**Comment by gen2brain** [comment]
Just define `-DPLATFORM_DESKTOP_SDL` and since you are on Linux, you can just use the output of `pkg-config --cflags --libs sdl2` when building.  I just tried your example with Go bindings, I do not get the warning though,

```
$ go build
$ ./t
ScreenWidth: 10, ScreenHeight: 10
RenderWidth: 10, RenderHeight: 10

$ go build -tags sdl
$ ./t
ScreenWidth: 400, ScreenHeight: 400
RenderWidth: 10, RenderHeight: 10
```

**Reply by Sqvid** [reply]
Thank you very much for your reply, and thank you for also testing my issue. Yes it does seem specific to my system. I am unsure how to track down the cause of the bug. I suppose it could be glfw, or my Wayland window manager (sway).

**Reply by gen2brain** [reply]
Well, for me this happens in the X11 session, with Openbox, so only the warning seems related to your WM but not the actual result. Also, note that GLFW builds by default for X11, you would need one build for `_GLFW_WAYLAND` and one for `_GLFW_X11`. I think only in the next version it will decide what to use on runtime.

**Reply by Sqvid** [reply]
I tried passing -DPLATFORM_DESKTOP_SDL=ON but it seems to be ignored by cmake. I am also passing -DGRAPHICS=GRAPHICS_API_SDL. any ideas?

CMake Warning:
  Manually-specified variables were not used by the project:

    PLATFORM_DESKTOP_SDL

**Comment by Peter0x44** [comment, marked as answer]
@Sqvid Raylib's CMake setup does not currently support building against SDL, you must use the Makefile in raylib/src.
My current strategy is to make the following changes to the Makefile:
```diff
-SDL_INCLUDE_PATH      ?= $(RAYLIB_SRC_PATH)/external/SDL2/include
-SDL_LIBRARY_PATH      ?= $(RAYLIB_SRC_PATH)/external/SDL2/lib
+SDL_INCLUDE_PATH      ?= $(shell pkg-config --cflags sdl2)
+SDL_LIBRARY_PATH      ?= $(shell pkg-config --libs sdl2)
-    INCLUDE_PATHS += -I$(SDL_INCLUDE_PATH)
+    INCLUDE_PATHS += $(SDL_INCLUDE_PATH)
-    LDFLAGS += -L$(SDL_LIBRARY_PATH)
+    LDFLAGS += $(SDL_LIBRARY_PATH)
```
then build with `make PLATFORM=PLATFORM_DESKTOP_SDL`, and also run `make clean` first, if necessary.
This should create a libraylib.a using SDL.

Then, you can build your code against it with:
`eval cc program.c -I path/to/raylib/src path/to/libraylib.a $(pkg-config --libs sdl2)`

**Reply by Sqvid** [reply]
Thank you very much. I was indeed able to compile raylib against SDL2 using your instructions. 

This does also bypass #3763, though I wish I didn't require this work around. I'll try and locate which glfw calls ought to be reported upstream.

**Reply by masroof-maindak** [reply]
> @Sqvid Raylib's CMake setup does not currently support building against SDL, you must use the Makefile in raylib/src. My current strategy is to make the following changes to the Makefile:
> 
> ```diff
> -SDL_INCLUDE_PATH      ?= $(RAYLIB_SRC_PATH)/external/SDL2/include
> -SDL_LIBRARY_PATH      ?= $(RAYLIB_SRC_PATH)/external/SDL2/lib
> +SDL_INCLUDE_PATH      ?= $(shell pkg-config --cflags sdl2)
> +SDL_LIBRARY_PATH      ?= $(shell pkg-config --libs sdl2)
> -    INCLUDE_PATHS += -I$(SDL_INCLUDE_PATH)
> +    INCLUDE_PATHS += $(SDL_INCLUDE_PATH)
> -    LDFLAGS += -L$(SDL_LIBRARY_PATH)
> +    LDFLAGS += $(SDL_LIBRARY_PATH)
> ```
> 
> then build with `make PLATFORM=PLATFORM_DESKTOP_SDL`, and also run `make clean` first, if necessary. This should create a libraylib.a using SDL.
> 
> Then, you can build your code against it with: `eval cc program.c -I path/to/raylib/src path/to/libraylib.a $(pkg-config --libs sdl2)`

Thank you for the response. What's stopping us from updating the repo's Makefile with this change? It seems quite sensible to me. I had to enable SDL2 because GLFW on Wayland is borked when HiDpi comes into play (exactly as in [here](https://github.com/raysan5/raylib/issues/3872)).

**Reply by Peter0x44** [reply]
@masroof-maindak the comment about cmake is no longer accurate, so you can use cmake if you want.

**Reply by masroof-maindak** [reply]
> @masroof-maindak the comment about cmake is no longer accurate, so you can use cmake if you want.

Gotcha. I'd like to stick with `make` for now though, and my question was w/ regards to it too. I could make a PR if that'd be something welcome.

**Reply by Peter0x44** [reply]
I remember there was some reason I didn't submit it
(possibly because it required removing the -L and breaking current setups)
This was over a year ago now so I don't remember the exact details. Feel free to look at it yourself.

**Comment by BrownNPC** [comment]
Why is this is still not supported by the CMakeLists.txt?

**Reply by Peter0x44** [reply]
I'm pretty sure it is. Maybe not in 5.0 though.

**Reply by BrownNPC** [reply]
can you point me to the CMake file, please?
